stdlib/header.c

```diff
@@ -93,9 +93,11 @@ static int file_ok(char * name)
 static char * searchpath(char * name)
 {
   char * path, * fullname, * p;
+  int len;
 
   path = getenv("PATH");
-  fullname = malloc(strlen(name) + (path == NULL ? 0 : strlen(path)) + 6);
+  len = (path == NULL ? 0 : strlen(path));
+  fullname = caml_stat_alloc_noexc(strlen(name) + len + 6);
```

Comment:
You have not included `memory.h` in this file (you can test either on Cygwin or temporarily hack `configure` so that the `#!` test fails which will cause this file to be linked).

However, this is the header for executables and the memory allocated is irrelevant after the `execv` call.